frontend/src/routes/Infrastructure/Clusters/ManagedClusters/CreateCluster/controlData/ControlDataHelpers.js

```diff
@@ -718,6 +724,45 @@ function versionGreater(version, x, y) {
   return matches && parseInt(matches[1], 10) >= x && parseInt(matches[2], 10) > y
 }
 
+export const addDeprecationWarningRHV = (controlData, t) => {
+  const infraIndex = controlData.findIndex(({ id }) => id === 'infrastructure')
+  if (controlData[infraIndex].active === 'RHV') {
+    const inx = controlData.findIndex(({ id }) => id === 'imageSet')
+    controlData.splice(inx, 1, {
+      name: t('cluster.create.ocp.image'),
+      tooltip: t('tooltip.cluster.create.ocp.image'),
+      id: 'imageSet',
+      type: 'combobox',
+      simplified: getSimplifiedImageName,
+      placeholder: t('creation.ocp.cloud.select.ocp.image'),
+      fetchAvailable: LOAD_OCP_IMAGES('rhv', t),
+      validation: {
+        notification: t('creation.ocp.cluster.must.select.ocp.image'),
+        required: true,
+      },
+      info: (_control, _controlData) => {
+        const imageSet = getControlByID(_controlData, 'imageSet')
+        if (imageSet && imageSet.active && imageSet.active.includes('4.13')) {
+          return (
+            <AcmAlert
+              variant="warning"
+              title={t('Deprecated host platform')}
+              message={t(
+                'Red Hat Virtualization is deprecated as a host platform for OpenShift 4.13 and will be removed in the next release.'
+              )}
+              noClose={true}
+              style={{ marginTop: '2em' }}
+            />
+          )
+        }
+        return {}
+      },
+    })
+  }
+
+  return controlData
+}
+
```

Comment:
I have tried to eliminate "controlData.splice" as much as possible. Side-effects are hard to follow, and it is inefficient because we are creating controlData, then searching through it to find the control we want and replacing it with a different version. You can see we've also duplicated code here.

Please take a look at the `getControlDataRHV` function. I think we can add this `info` prop directly there. There is a 3rd argument which will be `t`.

Reply:
Also, note we usually only start param names with underscore if we are not going to use those parameters but need to specify them because we are going to use positional parameters that come later in the list. So `_controlData` should not start with underscore.